docs/DeveloperGuide.md

```diff
@@ -1,34 +1,245 @@
+remainder to the user. It implements the following operations:
+* `RepaymentList#storeCurrentString()` → stores the repayment information to the repayment list
+* `Ui#printRepay(repaymentList.returnCurrentString()` → prints the repayment information which has been successfully stored
+
+Below shows an example of usage:
+1. User executes `repay Johnny SGD 5.0 2020-12-20` command to add the repayment information that one has to repay 
+SGD 5.0 to Johnny before 2020-12-20
+2. The `repay` command calls the `RepayCommand#execute(spendingList, ui)` to complete the adding process
+
+Figure below shows the sequence diagram of `RepayCommand` class.
+
+![Sequence Diagram of RepayCommand class](images/RepayCommand.png) </br>
+Figure 7: Sequence diagram of `RepayCommand` class
+
+### 3.7 Summary Feature
+`SpendingList` and `Item` facilitate this feature. The Summary feature is able to summarise the total amount spent 
+given a date. It implements the following operations:
+* `Item#getDate()` → gets the date when user spent on the item
+* `Item#getAmount()` → gets the amount spent on the item
+* `SpendingList#getSpendingAmountTime(period)` → gets the total amount spent during a period
+* `SpendingList#getSpendingAmountCategory(category, period)` → gets the total amount spent of a given category
+
+Below shows an example of usage:
+
+1. User executes the `summary 2020` command to get the amount spent during year 2020.
+2. The `summary` command calls the `SpendinList#getSpendingAmount(2020)` which checks the spending date of every items
+stored in the memory.
+3. If the item is spent during year 2020, the amount spent will be summed up.
+
+Figure below shows the sequence diagram of `SummaryCommand` class.
+
+![Sequence Diagram of SummaryCommand class](images/summary.png) </br>
+Figure 8: Sequence diagram of `SummaryCommand` class
+
+### 3.8 Reminder Feature
+`SpendingList` and `WarnCommand` facilitate this feature. The Reminder feature is able to provide the user about the 
+total expenditure of the current week, starting on Monday. It implements the following operations:
+* `SpendingList#getSpendingAmountTime(period)` 
+* `WarnCommand#execute(spendingList, ui)`
+
+Below shows an example of usage:
+
+1. User starts the application.
+2. The `Reminder` will be instantiated. The dates of the current week (starting from Monday) will be saved to a list.
+3. In the `Reminder#execute(spendinglist, ui)`, a check will be done to see if there is any budget being set by the user.
+    * If no budget is being set, the total expenditure of current week will be tallied up. 
+    * If there is, `WarnCommand#execute(spendingList, ui)` will be called first before tallying up the expenditure. 
+
+Figure below shows the sequence diagram of Reminder class.
+![Sequence Diagram of SummaryCommand class](images/reminder.png) </br>
+Figure 9: Sequence diagram of `SummaryCommand` class
```

Comment:
Should figure 9 be the sequence diagram of `Reminder` class instead as figure 8 already shows the sequence diagram of `SummaryCommand` class?